### PR TITLE
[docker]  event processing testing

### PR DESCRIPTION
**Pull request by xvello**
### What does this PR do?

Add unit testing for docker event processing

### CODE REVIEW

**Review by hkaj — APPROVED** [review]

File: pkg/util/docker/global.go [hunk 1]
```diff
@@ -60,6 +60,9 @@ func HostnameProvider(hostName string) (string, error) {
 
 // ContainerIDToEntityName returns a prefixed entity name from a container ID
 func ContainerIDToEntityName(cid string) string {
+	if cid == "" {
+		return ""
```
Comment (CharlyF) [inline pkg/util/docker/global.go]: Could we return an error instead ? 
It is called here https://github.com/DataDog/datadog-agent/blob/master/pkg/util/docker/events.go#L36 and then creates a tag https://github.com/DataDog/datadog-agent/blob/master/pkg/collector/corechecks/containers/docker_events.go#L65 but I could not find where the empty string would error out (https://github.com/DataDog/datadog-agent/blob/master/pkg/tagger/global.go#L24) as opposed to gathering all the tags from the empty cids.

**Review by CharlyF — COMMENTED** [review]
LGTM, just a quick question on the return if the cid is empty.

File: pkg/util/docker/global.go (hunk 1 above)
Comment (xvello) [reply pkg/util/docker/global.go]: My reasoning would be that adding that kind of check might drift to "why does this method not return an error if the container ID I give it is not running on that host?" As it's used for every dsd packet it has to be fast.

As "" is the default string empty value, it should indeed be ignored by the tagger (it represents an unknown/invalid entity). I'll add a check `if entity == "", return err` in tagger.Tag

Comment (xvello) [reply pkg/util/docker/global.go]: @CharlyF https://github.com/DataDog/datadog-agent/pull/734

Comment (CharlyF) [reply pkg/util/docker/global.go]: 👍

**Pull request merged**